python/cdp/actions/evm/swap/swap.py

```diff
@@ -0,0 +1,141 @@
+                    slippage_percentage=1.0
+                )
+            ),
+            swap_strategy=AccountSwapStrategy()
+        )
+        ```
+
+        **Using pre-created swap data**:
+        ```python
+        # First create the swap
+        swap_data = await api_clients.evm.create_swap(...)
+
+        # Then execute it
+        result = await swap(
+            api_clients=api_clients,
+            from_account=account,
+            swap_options=SwapOptions(create_swap_result=swap_data),
+            swap_strategy=AccountSwapStrategy()
+        )
+        ```
+
+    """
+    # Import EVM client here to avoid circular imports
+    from cdp.evm_client import EvmClient
+
+    # Create an EVM client instance
+    evm_client = EvmClient(api_clients)
+
+    # Determine which pattern is being used
+    if swap_options.create_swap_options:
+        # Pattern 1: SDK calls createSwap under the hood
+        create_options = swap_options.create_swap_options
+
+        # Get a quote for the swap
+        quote = await evm_client.get_quote(
+            from_token=create_options.from_token,
+            to_token=create_options.to_token,
+            amount=create_options.amount,
+            network=create_options.network,
+        )
+
+        # Create the swap
+        swap_tx = await evm_client.create_swap(
+            from_token=create_options.from_token,
+            to_token=create_options.to_token,
+            amount=create_options.amount,
+            network=create_options.network,
+            wallet_address=from_account.address,
+            slippage_percentage=create_options.slippage_percentage,
+        )
+
+        # Handle Permit2 signature if required
+        permit2_signature = None
+        if swap_tx.requires_signature and swap_tx.permit2_data:
+            # Sign the Permit2 typed data
+            typed_data = swap_tx.permit2_data.eip712
+            permit2_signature = await from_account.sign_typed_data(
+                domain=typed_data["domain"],
+                types=typed_data["types"],
+                primary_type=typed_data["primaryType"],
+                message=typed_data["message"],
+            )
+
+        # Convert SwapTransaction to CreateSwapResult
+        swap_data = CreateSwapResult(
+            quote_id=quote.quote_id,
+            from_token=create_options.from_token,
+            to_token=create_options.to_token,
+            from_amount=create_options.amount,
+            to_amount=quote.to_amount,
+            to=swap_tx.to,
+            data=swap_tx.data,
+            value=str(swap_tx.value),
+            gas_limit=None,  # Will be estimated during execution
+            gas_price=None,
+            max_fee_per_gas=None,
+            max_priority_fee_per_gas=None,
+        )
+
+        network = create_options.network
+    else:
+        # Pattern 2: Use pre-created swap data
+        swap_data = swap_options.create_swap_result
+        network = None  # Will be determined by strategy
```

Comment:
confused by network being None here but we also raise an Error in AccountSwapStrategy.execute_swap when network is None. Perhaps Im missing something?